### PR TITLE
feat: mint position

### PR DESCRIPTION
**Pull request by dannydaniil**

### CODE REVIEW

File: templates/react/src/example/Example.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { ethers } from 'ethers'
 import { AlphaRouter, ChainId, SwapType } from '@uniswap/smart-order-router'
 import { TradeType, CurrencyAmount, Percent } from '@uniswap/sdk-core'
 import { Environment, CurrentConfig } from '../config'
-import { getCurrencyBalance } from '../libs/wallet'
+import { getCurrencyBalance } from '../libs/balance'
```
Comment (adjkant) [inline templates/react/src/example/Example.tsx:7]: I was planning to add some other functions like wrapping/unwrapping ETH in some cases - can we keep this as wallet? The ABI rename makes sense still but we shouldn't keep this only to balance IMO

File: templates/react/src/libs/providers.ts [hunk 2]
```diff
@@ -109,12 +108,32 @@ async function sendTransactionViaExtension(
...
+  if (transaction.value) {
+    transaction.value = BigNumber.from(transaction.value)
+  }
+  const txRes = await wallet.sendTransaction(transaction)
+
+  let receipt = null
+  const provider = getProvider()
+  if (!provider) {
+    return TransactionState.Failed
+  }
```
Comment (adjkant) [inline templates/react/src/libs/providers.ts:120]: Should we check this first to fail fast?

File: templates/react/src/libs/providers.ts [hunk 3]
```diff
@@ -109,12 +108,32 @@ async function sendTransactionViaExtension(
 async function sendTransactionViaWallet(
   transaction: ethers.providers.TransactionRequest
 ): Promise<TransactionState> {
-  transaction.value = BigNumber.from(transaction.value)
-  const res = await wallet.sendTransaction(transaction)
-  const receipt = await res.wait()
+  if (transaction.value) {
+    transaction.value = BigNumber.from(transaction.value)
+  }
```
Comment (adjkant) [inline templates/react/src/libs/providers.ts:113]: Good catch!

File: templates/react/src/libs/providers.ts [hunk 4]
```diff
@@ -109,12 +108,32 @@ async function sendTransactionViaExtension(
...
+      }
+    } catch (e) {
+      console.log(`Receipt error:`, e)
+      break
+    }
+  }
 
   // Transaction was successful if status === 1
-  if (receipt.status === 1) {
+  if (receipt) {
```
Comment (adjkant) [inline templates/react/src/libs/providers.ts:136]: Why the update here? Should we remove the comment?

File: v3-sdk/minting-position/src/config.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  }
+}
+
+// Example Configuration
+
+export const CurrentConfig: ExampleConfig = {
+  env: Environment.WALLET_EXTENSION,
+  rpc: {
+    local: 'http://localhost:8545',
+    mainnet: 'https://mainnet.infura.io/v3/0ac57a06f2994538829c14745750d721',
```
Comment (adjkant) [inline v3-sdk/minting-position/src/config.ts:38]: Should be empty here in Github!

File: v3-sdk/minting-position/src/config.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    out: Token
+    fee: FeeAmount
+    liquidity: number
+  }
+}
+
+// Example Configuration
+
+export const CurrentConfig: ExampleConfig = {
+  env: Environment.WALLET_EXTENSION,
```
Comment (adjkant) [inline v3-sdk/minting-position/src/config.ts:35]: Default local?

File: v3-sdk/minting-position/src/example/Example.tsx [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+  useOnBlockUpdated(async (blockNumber: number) => {
+    refreshBalances()
+    setBlockNumber(blockNumber)
+  })
+
+  // Update wallet state given a block number
+  const refreshBalances = useCallback(async () => {
+    const provider = getProvider()
+    const address = getWalletAddress()
+    if (address && provider) {
```
Comment (adjkant) [inline v3-sdk/minting-position/src/example/Example.tsx:198]: style: check for nulls and return early instead? Saves indent space and fail fast conventions

File: v3-sdk/minting-position/src/libs/contracts.ts [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+      ...transaction,
+      from: fromAddress,
+    })
+  } catch (e) {
+    console.error(e)
+    return TransactionState.Failed
+  }
+
+  return TransactionState.Sent
+}
```
Comment (adjkant) [inline v3-sdk/minting-position/src/libs/contracts.ts:90]: These two feel very distinct from balance info - maybe a `positions.ts` library instead? `contracts`is a bit overgeneralized to me IMO

File: v3-sdk/minting-position/src/example/Example.tsx [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+    poolContract.liquidity(),
+    poolContract.slot0(),
+  ])
+
+  return {
+    liquidity,
+    sqrtPriceX96: slot[0],
+    tick: slot[1],
+  }
+}
```
Comment (adjkant) [inline v3-sdk/minting-position/src/example/Example.tsx:93]: I've combined these on mine into a single `getPoolInfo` function. I can see the argument for separating, but I'm curious how you feel about that combination. It's a lot shorter codewise (saves a duplicated `getPoolContract` call + multiple `Promise.all`'s) and we can explain that some properties are constant while other is state in guides / comments

**Review by adjkant — COMMENTED** [review]
LP code looks good! Gave a first pass but yet to run/test, will do that before a final merge. Otherwise small org things, some style, and default config!

File: v3-sdk/minting-position/src/libs/contracts.ts (hunk 8 above)
Comment (dannydaniil) [reply v3-sdk/minting-position/src/libs/contracts.ts:90]: I see! Technically, they are the same thing but use case wise, it does feel different

File: v3-sdk/minting-position/src/example/Example.tsx (hunk 9 above)
Comment (dannydaniil) [reply v3-sdk/minting-position/src/example/Example.tsx:93]: Sounds good

File: templates/react/src/example/Example.tsx (hunk 1 above)
Comment (dannydaniil) [reply templates/react/src/example/Example.tsx:7]: Addressed below.

File: templates/react/src/libs/providers.ts [hunk 10]
```diff
@@ -108,16 +108,16 @@ async function sendTransactionViaExtension(
...
   if (transaction.value) {
     transaction.value = BigNumber.from(transaction.value)
   }
   const txRes = await wallet.sendTransaction(transaction)
-
   let receipt = null
-  const provider = getProvider()
-  if (!provider) {
-    return TransactionState.Failed
-  }
```
Comment (adjkant) [inline templates/react/src/libs/providers.ts:121]: nit: remove space to group?

File: v3-sdk/minting-position/src/example/Example.tsx [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+} from '../libs/constants'
+
+const getPoolInfo = async (): Promise<{
+  token0: string
+  token1: string
+  fee: number
+  tickSpacing: number
+  sqrtPriceX96: ethers.BigNumber
+  liquidity: ethers.BigNumber
+  tick: number
```
Comment (adjkant) [inline v3-sdk/minting-position/src/example/Example.tsx:37]: Time to break this into an interface? `PoolInfo`?

File: v3-sdk/minting-position/src/libs/constants.ts [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+  // Authenticated Functions
+  'function transfer(address to, uint amount) returns (bool)',
+  'function approve(address _spender, uint256 _value) returns (bool)',
+
+  // Events
+  'event Transfer(address indexed from, address indexed to, uint amount)',
+]
+
+export const NONFUNGIBLEPOSITIONMANAGER_ABI = [
```
Comment (adjkant) [inline v3-sdk/minting-position/src/libs/constants.ts:52]: naming: `NONFUNGIBLE_POSITION_MANAGER_ABI`?

**Review by adjkant — COMMENTED** [review]
LGTM with two naming updates / a spacing nit!

File: v3-sdk/minting-position/src/example/Example.tsx (hunk 11 above)
Comment (dannydaniil) [reply v3-sdk/minting-position/src/example/Example.tsx:37]: I intentionally did not pull this out, to not complicate things too much but I guess I'm good with it.

**Review by adjkant — APPROVED** [review]
Ship it!

**Pull request merged**